designsafe/static/scripts/ng-designsafe/services/file-listing-service.js

```diff
@@ -0,0 +1,632 @@
+        )
+        .subscribe();
+    return subject;
+};
+
+export class FileListingService {
+    constructor(
+        $http,
+        $uibModal,
+        $rootScope,
+        $q,
+        ProjectService,
+        ProjectEntitiesService,
+        FileOperationService,
+        Django
+    ) {
+        'ngInject';
+        this.count = 0;
+        this.$rootScope = $rootScope;
+        this.$uibModal = $uibModal;
+        this.Django = Django;
+        this.$q = $q;
+        this.$http = $http;
+        this.ProjectService = ProjectService;
+        this.FileOperationService = FileOperationService;
+        this.ProjectEntitiesService = ProjectEntitiesService;
+        this.from = from; // bind rxjs method for mocking
+
+        this.modalCloseSubject = new Subject();
+        this.listingStartSubject = new Subject();
+        this.abstractListingSubject = takeLatestSubscriber();
+
+        this.listings = {
+            main: {
+                loading: false,
+                loadingScroll: false,
+                reachedEnd: false,
+                listing: [],
+                selectedFiles: [],
+                selectAll: false,
+                params: {
+                    api: '',
+                    system: '',
+                    scheme: 'private',
+                    path: '',
+                    offset: 0,
+                    limit: 100,
+                },
+                listingSubscriber: takeLatestSubscriber(),
+                scrollSubscriber: takeLeadingSubscriber(),
+                error: null,
+                errorScroll: null,
+                rootEntities: [],
+                selectedForPublication: false,
+            },
+            modal: {
+                loading: false,
+                loadingScroll: false,
+                reachedEnd: false,
+                listing: [],
+                selectedFiles: [],
+                selectAll: false,
+                params: {
+                    api: '',
+                    system: '',
+                    scheme: 'private',
+                    path: '',
+                    offset: 0,
+                    limit: 100,
+                },
+                listingSubscriber: takeLatestSubscriber(),
+                scrollSubscriber: takeLeadingSubscriber(),
+                error: null,
+                errorScroll: null,
+                selectedForPublication: false,
+            },
+        };
+
+        this.selectedListing = null; //Listing of selected files in publication pipeline
+
+        // Map legacy fileMgr values to api/scheme pair and breadcrumb settings.
+        this.fileMgrMappings = {
+            agave: {
+                api: 'agave',
+                scheme: 'private',
+                breadcrumbParams: {
+                    skipRoot: true,
+                    customRoot: { label: 'My Data', path: this.Django.user },
+                },
+            },
+            shared: {
```

Comment:
I spoke too soon.  I have an empty "shared" file listing on master, but on this branch I get a 500 on the api call.  "listing() missing 1 required positional argument: 'username'"